wire-library/wire-runtime-swift/src/test/swift/ProtoReaderTests.swift

```diff
@@ -332,6 +443,72 @@ final class ProtoReaderTests: XCTestCase {
         }
     }
 
+    func testDecodePackedRepeatedUnknownEnumsSomeFail() throws {
+        let data = Data(hexEncoded: """
+            0A       // (tag 1 | Length Delimited)
+            02       // Number of enums
+            01       // .HOME
+            05       // Unknown enum
+        """)!
+
+        try test(data: data, enumStrategy: .returnNil) { reader in
+            var values: [Person.PhoneType] = []
+
+            let fields = try reader.forEachTag { tag in
+                switch tag {
+                case 1:
+                    try reader.decode(into: &values)
+                default:
+                    XCTFail("Should not encounter unknown fields")
+                }
+            }
+
+            let expectedData = Data(hexEncoded: """
+                       08       // (Tag 1 | Varint)
+                       05       // Unknown enum
+                   """)!
+
+            XCTAssertEqual(values, [.HOME])
+            XCTAssertEqual(fields, expectedData)
+        }
+    }
+
+    func testDecodePackedRepeatedUnknownEnumsAllFail() throws {
+        let data = Data(hexEncoded: """
+            0A       // (tag 1 | Length Delimited)
+            02       // Number of enums
+            04       // Unknown enum
+            05       // Unknown enum
+        """)!
+        try test(data: data, enumStrategy: .returnNil) { reader in
+            var values: [Person.PhoneType] = []
+
+            let fields = try reader.forEachTag { tag in
+                switch tag {
+                case 1:
+                    try reader.decode(into: &values)
+                default:
+                    XCTFail("Should not encounter unknown fields")
+                }
+            }
+
+            XCTAssertEqual(values, [])
+            
+            // The original data is packed (length delimited), but we encode it as
+            // unpacked in the unknown data, so it shows up as individual varints
+            // So format changes from `0A_02_04_05` to `08_04_08_05`.
```

Comment:
nit: period after "varints" and the "So the format changes" like feels superfluous since it's in the code.